src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.InlineContext.cs

```diff
@@ -0,0 +1,601 @@
+                .DescendantNodesAndSelf()
+                .Where(node => node != null && _syntaxFacts.IsAwaitExpression(node))
+                .ToImmutableArray();
+            foreach (var awaitExpression in awaitExpressions)
+            {
+                var enclosingMethodLikeNode = GetEnclosingMethodLikeNode(awaitExpression);
+                if (calleeMethodDeclarationNode.Equals(enclosingMethodLikeNode))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private SyntaxNode CreateLocalDeclarationStatement(
+            SyntaxGenerator syntaxGenerator,
+            ImmutableDictionary<ISymbol, string> renameTable,
+            IArgumentOperation argumentOperation)
+        {
+            var parameterSymbol = argumentOperation.Parameter;
+            var name = renameTable.TryGetValue(parameterSymbol, out var newName) ? newName : parameterSymbol.Name;
+            // Check has been done before to make sure there is only one child.
+            var argumentExpressionOperation = argumentOperation.Children.Single();
+            if (argumentOperation.ArgumentKind == ArgumentKind.ParamArray
+                && parameterSymbol.Type is IArrayTypeSymbol paramArrayParameter
+                && argumentExpressionOperation is IArrayCreationOperation arrayCreationOperation
+                && argumentOperation.IsImplicit)
+            {
+                // if this argument is a param array & the array creation operation is implicitly generated,
+                // it means it is in this format:
+                // void caller() { Callee(1, 2, 3); }
+                // void Callee(params int[] x) { }
+                // Collect each of these arguments and generate a new array for it.
+                // Note: it could be empty.
+                return syntaxGenerator.LocalDeclarationStatement(
+                    paramArrayParameter,
+                    name,
+                    syntaxGenerator.ArrayCreationExpression(GenerateTypeSyntax(paramArrayParameter.ElementType, allowVar: false), arrayCreationOperation.Initializer.ElementValues.SelectAsArray(op => op.Syntax)));
+            }
+            else
+            {
+                // In all the other cases, one parameter should only maps to one arguments.
+                return syntaxGenerator.LocalDeclarationStatement(parameterSymbol.Type, name, argumentExpressionOperation.Syntax);
+            }
+        }
+
+        private static async Task<TExpressionSyntax> ReplaceAllSyntaxNodesForSymbolAsync(
+            Document document,
+            TExpressionSyntax inlineExpression,
+            SyntaxGenerator syntaxGenerator,
+            ImmutableDictionary<ISymbol, SyntaxNode> replacementTable,
+            CancellationToken cancellationToken)
+        {
+            var editor = new SyntaxEditor(inlineExpression, syntaxGenerator);
+
+            foreach (var kvp in replacementTable)
```

Comment:
Put a comments here because I get this error when using deconstruct method
![image](https://user-images.githubusercontent.com/24360909/91211031-f5329c00-e6c2-11ea-992a-c24513fff296.png)

Reply:
its' an extension method.  you need to add an appropriate `using` statement to have it be available.